ckanext/aberdeen/actions.py

```diff
@@ -179,3 +182,69 @@ def group_purge(context, data_dict):
 
 def organization_purge(context, data_dict):
     return _group_or_org_purge(context, data_dict, is_org=True)
+
+
+@toolkit.side_effect_free
+def inactive_users(context, data_dict):
+    '''Returns a list of users that have been inactive for the last year'''
+
+    if not authz.is_sysadmin(toolkit.c.user):
+        toolkit.abort(403, _('You are not authorized to access this list'))
+
+    user_list = ckan.logic.get_action('user_list')(context, data_dict)
+    inactive_users = []
+    threads = []
+
+    # Limit the user activity results to the most recent event
+    data_dict['limit'] = 1
+
+    # Use threading to speed up the inactive user collection
+    def user_activity_threads(user, inactive_users):
+        data_dict['id'] = user['id']
+        user_info = ckan.logic.get_action(
+            'user_activity_list')(context, data_dict)
+
+        if not user_info:
+            return
+
+        timestamp = datetime.strptime(
+            user_info[0]['timestamp'], '%Y-%m-%dT%H:%M:%S.%f')
+
+        if timestamp < datetime.today() - timedelta(days=365):
+            user['last_activity'] = user_info[0]['timestamp']
+            inactive_users.append(user)
+
+        return inactive_users
+
+    for user in user_list:
+        thread = threading.Thread(
+            target=user_activity_threads, args=(user, inactive_users))
+        threads.append(thread)
+        thread.start()
+
+    for user, thread in enumerate(threads):
+        thread.join()
+
+    return inactive_users
+
+
+@toolkit.side_effect_free
+def send_inactive_users_email(context, data_dict):
+    '''Sends the inactive users list to all site system administrators'''
+
+    user_list = ckan.logic.get_action('user_list')(context, data_dict)
+    admin_email_list = [
+        user['email'] for user in user_list if user['sysadmin'] is True]
+
+    inactive_users = toolkit.get_action('inactive_users')(context, data_dict)
+
+    # Format the inactive users list for better readability in an email
+    inactive_users = '\n\n'.join(
+        '\n'.join(info + ': ' + str(user_info[info]) for info in user_info)
+        for user_info in inactive_users)
```

Comment:
Can we prepare more meaning full text? Something like
```
Hello <<Admin Name>>,

Sending a list of inactive users.

- User Name - https://data.aberdeencity.gov.uk/user/<<username>>
- User name...
```

Reply:
Yes! This is something I forgot to ask about. I did make the email subject descriptive, but the emails seem like they need more in them.

Reply:
@zelima Do you think it would be better to still send an email if there are no inactive users?
```
Hi <ADMIN>,

There are no inactive users this week.
```
Or just skip the emails in that case?

Reply:
I changed the email output. This is what it looks like now: 

![Screenshot_20200108-123926](https://user-images.githubusercontent.com/19521691/72001822-2df53200-3214-11ea-82d6-01de960a3e5c.jpg)